src/languages/javascript.js

```diff
@@ -90,6 +90,10 @@ export default function(hljs) {
     hljs.REGEXP_MODE
   ];
   var PARAMS_CONTAINS = SUBST.contains.concat([
+    // eat recursive parens in sub expressions
+    { begin: /\(/, end: /\)/,
+      contains: ["self", hljs.C_LINE_COMMENT_MODE,hljs.C_BLOCK_COMMENT_MODE]
+    },
```

Comment:
What is the purpose of this "parens eater"? Due to this, some highlighting is lost inside params.
E. g. for this snippet:
```
sides.every((length,width=(3+2+(4/5))) => length > 0 );
```
we had before:
```
sides.every(<span class="hljs-function">(<span class="hljs-params">length,width=(
<span class="hljs-number">3</span>+<span class="hljs-number">2</span>+(<span class="hljs-number">4</span>/<span class="hljs-number">5</span></span>))
) =&gt;</span> length &gt; <span class="hljs-number">0</span> );
```
and after the patch we have:
```
sides.every(<span class="hljs-function">(<span class="hljs-params">length,width=(
3+2+(4/5))</span>
) =&gt;</span> length &gt; <span class="hljs-number">0</span> );
```
I. e. numbers are no more highlighted.
(In typescript, numbers are not highlighted even now)

Reply:
We have to count parens to know which one is the proper one to stop at... but this issue you pointed out can be resolved by just adding more things to the params ruleset. I'll look into that.

Reply:
https://github.com/highlightjs/highlight.js/pull/2372#issuecomment-583271131

Reply:
?

Reply:
@yyyc514 It is okay. It&rsquo;s just that I know this issue will come up as I thought about the function argument pattern problem, like in the comment I have made there.

Reply:
This fix handles both those cases without issue.

Reply:
@egor-rogov Have another look please now.